godtools/App/Features/ToolDetails/Presentation/ToolDetails/ToolDetailsViewModel.swift

```diff
@@ -0,0 +1,252 @@
+                    
+                case .failure(let error):
+                    
+                    hidesLearnToShareToolButtonValue = true
+                }
+                
+                self?.hidesLearnToShareToolButton = hidesLearnToShareToolButtonValue
+            }
+        })
+    }
+    
+    private func reloadFavorited() {
+                
+        if favoritedResourcesCache.isFavorited(resourceId: resource.id) {
+            hidesAddToFavoritesButton = true
+            hidesRemoveFromFavoritesButton = false
+        }
+        else {
+            hidesAddToFavoritesButton = false
+            hidesRemoveFromFavoritesButton = true
+        }
+    }
+    
+    private func reloadToolDetails() {
+        
+        let resourcesCache: ResourcesCache = dataDownloader.resourcesCache
+
+        let nameValue: String
+        let aboutDetailsValue: String
+        let languageBundle: Bundle
+        
+        if let primaryLanguage = languageSettingsService.primaryLanguage.value, let primaryTranslation = resourcesCache.getResourceLanguageTranslation(resourceId: resource.id, languageId: primaryLanguage.id) {
+            
+            nameValue = primaryTranslation.translatedName
+            aboutDetailsValue = primaryTranslation.translatedDescription
+            languageBundle = localizationServices.bundleLoader.bundleForResource(resourceName: primaryLanguage.code) ?? Bundle.main
+        }
+        else if let englishTranslation = resourcesCache.getResourceLanguageTranslation(resourceId: resource.id, languageCode: "en") {
+            
+            nameValue = englishTranslation.translatedName
+            aboutDetailsValue = englishTranslation.translatedDescription
+            languageBundle = localizationServices.bundleLoader.englishBundle ?? Bundle.main
+        }
+        else {
+            
+            nameValue = resource.name
+            aboutDetailsValue = resource.resourceDescription
+            languageBundle = localizationServices.bundleLoader.englishBundle ?? Bundle.main
+        }
+                
+        name = nameValue
+        totalViews = String.localizedStringWithFormat(localizationServices.stringForBundle(bundle: languageBundle, key: "total_views"), resource.totalViews)
+        openToolButtonTitle = localizationServices.stringForBundle(bundle: languageBundle, key: "toolinfo_opentool")
+        learnToShareToolButtonTitle = localizationServices.stringForBundle(bundle: languageBundle, key: "toolDetails.learnToShareToolButton.title")
+        addToFavoritesButtonTitle = localizationServices.stringForBundle(bundle: languageBundle, key: "add_to_favorites")
+        removeFromFavoritesButtonTitle = localizationServices.stringForBundle(bundle: languageBundle, key: "remove_from_favorites")
+        aboutTitle = localizationServices.stringForBundle(bundle: languageBundle, key: "about")
+        versionsTitle = localizationServices.stringForBundle(bundle: languageBundle, key: "toolDetails.versions.title")
```

Comment:
What do you think about having a container to store all the string keys?  I tend to think having the literals is error-prone, and difficult to change if something needs to be adjusted in the future.  Something like
```
enum LocalizedStringKey {
  static let totalViews = "total_views"
  static let openToolButtonTitle = "toolinfo_opentool"
 ...etc...
}
```

Reply:
Hey @rachaelblue created a JIRA task for this one here: https://jira.cru.org/browse/GT-1621